index.bs

```diff
@@ -2153,6 +2158,82 @@ It is recommended that once [=explicit consent=] is granted for a specific [=/or
+
+### Rounding, quantization, and fuzzing ### {#rounding-and-friends}
+Rounding, quantization, and fuzzing are three categories of mitigations that modify the raw data that would otherwise be returned to the developer. Rounding decreases the precision of data by reducing the number of digits used to express it. Quantization constrains continuous data to instead report a discrete subset of values. Fuzzing is the introduction of slight, random errors into the the data. Collectively, these mitigations are useful to avoid fingerprinting, and are especially useful when doing so does not cause noticeable impact on user comfort.
+
+### Limiting ### {#limiting}
+Limiting is when data is reported only when it is within a specific range. For example, it is possible to comfortably limit reporting positional pose data when a user has moved beyond a specific distance away from an approved location. Care should be taken to ensure that the user experience is not negatively affected when employing this mitigation. It is often desireable to avoid a 'hard stop' at the at the end of a range as this may cause disruptive user experiences.
+
+Protected functionality {#protected-functionality}
+-----------------------
+
+The [=sensitive information=] exposed by the API can be divided into categories that share threat profiles and necessary protections against those threats.
+
+### Immersiveness ### {#protect-immersiveness}
+Users must be in control of when immersive sessions are created because the creation causes invasive changes on a user's machine. For example, starting an [=immersive session=] will engage the [=/XR device=] sensors, take over access to the device's display, and begin presentating immersive content which may terminate another application's access to the XR hardware. It may also incur significant power or performance overhead on some systems or trigger the launching of a status tray or storefront.
+
+<div class="algorithm" data-algorithm="immersive-session-allowed">
+
+To determine if an <dfn>immersive session request is allowed</dfn> the user agent MUST run the following steps:
+
+  1. If the request was not [=triggered by user activation=], return <code>false</code>
+  1. If the requesting document is not considered [=trustworthy=], return <code>false</code>
+  1. If [=user intent=] to begin an [=immersive session=] is not well understood, either via [=explicit consent=] or [=implicit consent=], return <code>false</code>
+  1. Return <code>true</code>
+
+</div>
+
+Starting an {{XRSessionMode/"inline"}} session does not implicitly carry the same requirements, though additional requirements may be imposed depending on the session's [=requested features=].
+
+<div class="algorithm" data-algorithm="inline-session-allowed">
+
+To determine if an <dfn>inline session request is allowed</dfn> the user agent MUST run the following steps:
+
+  1. If the requesting document is not [=responsible=], return <code>false</code>
```

Comment:
As written this will allow inline sessions to trigger consent prompts without a user interaction.  Since that might slip through the cracks I'd like to confirm it is intentional. For context, the new api to prompt for access to accelerometer data was added since a bunch of sites already silently sniff this data.  However camera/mic access doesn't appear to have a interaction requirement.

Reply:
How would we want to frame that? First impression is to say that if any of the required features will require explicit consent and the request was not triggered by user activation then reject, but in order for that to be predictable it assumes that all browsers consider the same features to require explicit consent, and I don't think that's practical. Another approach would be to say that if _any_ optional or required features are present then user activation is required, but that feels overly aggressive. Thoughts?

Reply:
I'm not actually sure we'd even want to make a change here.  For example, camera/microphone don't require a user interaction before prompting.  Neither do notifications (though ugh that's sort of painful).

If we *did* decide to do this, the safest approach might just be to say we require a user action if any features are requested.  Alternatively, developers could call requestSession and only add a button to cause a user action if the initial call fails with a specific error.

Reply:
> the safest approach might just be to say we require a user action if any features are requested.

Thinking this through: In an ideal world developers could throw up inline content with no sensor data immediately, and then at some point in the future when the user chooses to allow it they could upgrade in-place to start using sensors. I don't really see how that exact pattern could work out with how the API is structured, though (and have no desire to restructure it to work around this one issue.)

The problem is that even if you specify a feature as optional you'd have to either have consent granted or denied before the session was created, which would block the inline session in the "perfect" scenario. And even if you somehow proceeded without waiting for consent most apps would query the `local` space immediately, fail, and never check again. You could build out features for allowing consent to be async but... No.

So it seems like the "right" developer pattern here is to create a viewer-only inline session right away, then wait for consent to be granted for sensor data and, if granted, swap out the sessions at the point that the sensor-enabled session is created. That pattern works with either a immediate prompt via a butter bar or similar OR waiting on user interaction. One of them is far less annoying to the user, though, so I'd probably prefer to have any features require user interaction. Plus it's a pattern that's well-reflected in the native world where apps like Google Maps have little icons in their 360 views to toggle accelerometer-based rotation on and off. (It's also an easier restriction to loosen in the future, whereas adding it after the fact would be a significant breaking change.)

Reply:
Works for me